### DISCUSSION TITLE
Fan Manual/Auto mode only in main accessory service

### DISCUSSION THREAD

**Discussion by michaelflux** [merdok/homebridge-miot#528] [Q&A] [answered]
@merdok wanted to get your input on what the recommended configuration would be.

Currently when a purifier is added to the plugin (tested on zhimi.airpurifier.mb3 and zhimi.airp.sa4), as expected all accessory sensors appear - temperature, air quality, humidity etc within HomeKit.

<img width="540" alt="Screenshot 2023-09-19 at 19 52 01" src="https://github.com/merdok/homebridge-miot/assets/2682413/1f79dd6f-a63a-4352-b03d-ee72d5871a3a">

If "Show only the main accessory service" is enabled, as expected that causes all accessory sensors to disappear, but the side effect, is that the Manual/Auto toggle+slider appears (which is what I want)

<img width="551" alt="Screenshot 2023-09-19 at 19 55 19" src="https://github.com/merdok/homebridge-miot/assets/2682413/d036484e-abdb-4788-abff-5f7b45cddbff">

What is the recommended way of retaining all accessory sensors, but allowing for the auto/manual toggle+slider (without resorting to creating a bunch of separate switches)? Manually adding property monitors seems to get rid of the toggle.

**Comment by merdok** [comment]
I am not certain but that sounds like bug. If I understand the issue correctly, I am not sure why the manual/auto and slider should only appear when the only main accessory service is enabled. I cannot validate that currently as I am traveling abroad. 
Could you maybe post a config from both scenarios?

Also please note that with the default behavior HomeKit is automatically creating an accessory group which contains the sensors and the actual air purifier device, you have to navigate inside the group to see the actual air purifier accessory. You can manually unground the accessories if needed. Maybe this is the issue for you?

**Reply by michaelflux** [reply]
With sensors appearing, but not manual/auto
```
{
                    "name": "Living Room Purifier 4 MAX",
                    "ip": "10.0.0.127",
                    "token": "************",
                    "deviceId": "************",
                    "model": "zhimi.airp.sa4",
                    "pollingInterval": 10,
                    "deepDebugLog": false,
                    "customAccessory": false,
                    "onlyMainService": false,
                    "buzzerControl": false,
                    "ledControl": false,
                    "childLockControl": false,
                    "modeControl": false,
                    "fanLevelControl": false,
                    "ioniserControl": false,
                    "screenControl": false
                }
```

With manual/auto, but no sensors.
```
{
                    "name": "Living Room Purifier 4 MAX",
                    "ip": "10.0.0.127",
                    "token": "************",
                    "deviceId": "************",
                    "model": "zhimi.airp.sa4",
                    "pollingInterval": 10,
                    "deepDebugLog": false,
                    "customAccessory": false,
                    "onlyMainService": true,
                    "buzzerControl": false,
                    "ledControl": false,
                    "childLockControl": false,
                    "modeControl": false,
                    "fanLevelControl": false,
                    "ioniserControl": false,
                    "screenControl": false
                }
```

The only difference is onlyMainService being true.

when in deepDebug, onlyMainService has no impact on any properties that are being output, identical in both configurations.

```
[
  "air-purifier:on: true",
  "air-purifier:fault: 0",
  "air-purifier:mode: 0",
  "air-purifier:fan-level: 2",
  "environment:relative-humidity: 49",
  "environment:air-quality: 1",
  "environment:pm2.5-density: 14",
  "environment:temperature: 28",
  "environment:hcho-density: 0 | (not synced)",
  "filter:filter-life-level: 91",
  "filter:filter-left-time: 6342",
  "filter:filter-used-time: 658",
  "alarm:alarm: true",
  "alarm:volume: 0 | (not synced)",
  "screen:brightness: 1",
  "physical-controls-locked:physical-controls-locked: false",
  "filter10:filter-life-level: 0 | (not synced)",
  "filter10:filter-left-time: 0 | (not synced)",
  "filter10:filter-used-time: 0 | (not synced)",
  "air-purifier-favorite:fan-level: 1",
  "habit:real-motor-feetback: 0 | (not synced)",
  "rfid:left-rfid-tag:  | (not synced)",
  "rfid:left-rfid-factory-id:  | (not synced)",
  "rfid:left-product-id:  | (not synced)",
  "rfid:left-rfid-time:  | (not synced)",
  "rfid:left-rfid-serial-num:  | (not synced)",
  "rfid:right-rfid-tag:  | (not synced)",
  "rfid:right-rfid-fac-id:  | (not synced)",
  "rfid:right-product-id:  | (not synced)",
  "rfid:right-rfid-time:  | (not synced)",
  "rfid:right-rfid-ser-num:  | (not synced)",
  "custom-service:favorite-square:  | (not synced)",
  "custom-service:form-sensor-sn:  | (not synced)",
  "aqi:aqi-updata-heartbeat: 0 | (not synced)"
]

```

Can confirm that when main accessory service enabled, sensors do not appear anywhere in HomeKit.

**Reply by merdok** [reply, marked as answer]
Sensors are not part of the main accessory service. Only the air purifier device is, so that is the correct behavior.

Please make sure that you are in the correct device(accessory) view, because in your first post on the images you are comparing an accessory group to a single accessory. The mode selection will not appear on the group but only on the accessory itself.

**Reply by michaelflux** [reply]
Ah right!  Yes yes, you're completely correct.  I had a dumb moment there and completely slipped my mind that the fan was being considered as an accessory within the group and not the main device itself.  Alright all works as expected then, thanks for the input on this!
